pkgs/by-name/an/ani-skip/package.nix

```
@@ -0,0 +1,55 @@
+{
+  stdenvNoCC,
+  fetchFromGitHub,
+  makeWrapper,
+  gnugrep,
+  gnused,
+  curl,
+  fzf,
+  lib,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "ani-skip";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "synacktraa";
+    repo = "ani-skip";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-VEEG3d6rwTAS7/+gBKHFKIg9zFfBu5eBOu6Z23621gM=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  runtimeInputs = [
+    gnugrep
+    gnused
+    curl
+    fzf
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D skip.lua $out/share/mpv/scripts/skip.lua
+    install -Dm 755 ani-skip $out/bin/ani-skip
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    substituteInPlace $out/bin/ani-skip \
+      --replace-fail '--script-opts=%s' "--script=$out/share/mpv/scripts/skip.lua --script-opts=%s"
+
+    wrapProgram $out/bin/ani-skip \
+      --suffix PATH : ${lib.makeBinPath finalAttrs.runtimeInputs}
```

```suggestion
      --suffix PATH : ${lib.makeBinPath [ gnugrep gnused curl fzf ]}
```
grep and sed are defaults so we don't necessarily need them

---

What's the point of this change? Why not just have them as an attribute for flexibility with overrides?

---

Very nit but this is what id do:

```
{
  ...
  extraRuntimeInputs ? [],
}:
stdenv.mkDerivation {
  ...

  wrapProgram --suffix PATH ${lib.makeBinPath ([ curl fzf ] ++ extraRuntimeInputs)}
  ...
}
```

Very nit and opinionated, the way youve done it is in my opinion perfectly acceptable

---

I'd rather keep the current implementation. Introducing an argument is overkill.

---

Actually this is better than what you suggests, because it allows overriding the list easily. Also we can't guarantee our programs don't run on other systems and external `grep` or `sed` is a source of impurity.

---

I also think the current approach is fine. @SuperSandro2000 Do you have any objections to merging this as is.